File: spring-boot-encoding-filter/src/main/java/com/example/controller/FormController.java
package com.example.controller;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;

import javax.annotation.Resource;

/**
 * Created by yamashiro-r on 2017/07/08.
 */
@Controller
@RequestMapping("/form")
public class FormController {
    @Resource
    private ShiftJisController shiftJisController;

    @GetMapping
    String form() {
        return "form";
    }

    @PostMapping
    String post(@ModelAttribute("text") String text, Model model) {
        ShiftJisController.ResponseAdaptor responseAdaptor = this.shiftJisController.postForm(text);
        model.addAttribute("showResponse", true);
        model.addAttribute("response", responseAdaptor);
        return "form";
    }

}

File: spring-boot-kotlin/src/main/kotlin/sample/spring/boot/controller/form/FormController.kt
package sample.spring.boot.controller.form

import org.slf4j.LoggerFactory
import org.springframework.http.MediaType
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.*
import org.springframework.web.multipart.MultipartFile
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import javax.validation.Valid

/**
 * Created by yamashiro-r on 2017/02/21.
 */
@Controller
@RequestMapping("/form/material")
class FormController {
    val logger = LoggerFactory.getLogger(FormController::class.java)

    @ModelAttribute("form")
    fun form() = MaterialForm(
            name = "picotaro",
            textarea = "pen pineapple apple pen",
            datepicker = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy/MM/dd")),
            hour = LocalDateTime.now().format(DateTimeFormatter.ofPattern("hh"))
    )

    @GetMapping
    fun material() = "form/material_form"

    @PostMapping(consumes = arrayOf(MediaType.MULTIPART_FORM_DATA_VALUE))
    fun post(@RequestParam("file") file: MultipartFile,
             @ModelAttribute("form") @Valid form: MaterialForm,
             model: Model): String {
        logger.info("post data: $form")
        model.addAttribute("form", form)
        return "form/material_form"
    }
}
